File: data/EXP01/python/2442/2442.output_1.py
# @lc app=leetcode id=2442 lang=python3
class Solution:
    def countDistinctIntegers(self, nums: List[int]) -> int:
        distinct_integers = set(nums)
        
        for num in nums:
            reversed_num = int(str(num)[::-1])
            distinct_integers.add(reversed_num)
        
        return len(distinct_integers)

File: data/EXP01/golang_no_system_var/2442/2442.output_1.go
func countDistinctIntegers(nums []int) int {
    distinctIntegers := make(map[int]bool)
    
    for _, num := range nums {
        distinctIntegers[num] = true
        
        reverseNum := reverseDigits(num)
        distinctIntegers[reverseNum] = true
    }
    
    return len(distinctIntegers)
}

func reverseDigits(num int) int {
    reversed := 0
    
    for num > 0 {
        digit := num % 10
        reversed = reversed*10 + digit
        num /= 10
    }
    
    return reversed
}